core/mm3Types.js

```diff
@@ -65,10 +163,61 @@ CLASS({
       help: 'Regex pattern for property.'
     },
     {
+      name: 'minChars',
+      label: 'Minimum characters',
+      help: 'The minimum number of characters required.'
+    },
+    {
+      name: 'maxChars',
+      label: 'Maximum characters',
+      help: 'The maximum number of characters allowed.'
+    },
+    {
       name: 'prototag',
       label: 'Protobuf tag',
       required: false,
       help: 'The protobuf tag number for this field.'
+    },
+    {
+      name: 'validate',
+      lazyFactory: function() {
+        var prop = this; // this == the property
+        var ret = constantFn('');
+
+        var min = prop.minChars;
+        if ( typeof min !== 'string' || min ) { // ignore '' (falsey string)
```

Comment:
Good point, I forgot about default values.  I still don't see why you need to parse it as an int.  You're not parsing it in NumericProperty_.

These should be consistent, and just checking if ( max !== '' ) seems to capture the condition better more obviously.

If max is empty string, then we don't validate max, otherwise we assume its an int and validate.  parseInt could return NaN which would really mess up validation.